File: Aditya Vardhan Singh/Other Problems/Day 15/Remove duplicates from a sorted doubly linked list/Python/Remove duplicates from a sorted doubly linked list.py
#Back-end complete function Template for Python 3

'''
# Node Class
        class Node:
            def __init__(self, data):   # data -> value stored in node
                self.data = data
                self.next = None
                self.prev = None
'''
class Solution:
    #Function to remove duplicates from unsorted linked list.
    def removeDuplicates(self, head):
        # code here
        # return head after editing list
        temp = head
        
        while temp.next:
            if temp.data == temp.next.data:
                temp.next = temp.next.next
            else:
                temp = temp.next
        
        return head
            

File: Aditya Vardhan Singh/Other Problems/Day 15/Remove duplicates from a sorted doubly linked list/C++/Remove duplicates from a sorted doubly linked list.cpp
/*
struct Node
{
    int data;
    Node * next;
    Node * prev;
    Node (int x)
    {
        data=x;
        next=NULL;
        prev=NULL;
    }
        
};
*/

class Solution
{
public:

    Node * removeDuplicates(struct Node *head)
    {
        if (!head->next) {
            return head;
        }
        
        // Your code here
        Node *temp = head;
        Node *del = nullptr;
        
        while (temp->next) {
            if (temp->data == temp->next->data) {
                del = temp->next;
                temp->next = temp->next->next;
                if (temp->next) {
                    temp->next->prev = temp;
                }
                del->next = del->prev = nullptr;
                delete(del);
            } else {
                temp = temp->next;
            }
        }
        
        return head;
    }
};